MeshLib/MeshSearch/ElementSearch.h

```diff
@@ -13,7 +13,10 @@
 #include <limits>
 #include <vector>
 
+#include <boost/optional.hpp>
```

Comment:
Do you have to include that manually? Isn't that implicitly done from the definition of `_mesh.getProperties().getPropertyVector<PROPERTY_TYPE>()` ?